src/app/store/trips/selectors.ts

```diff
@@ -0,0 +1,69 @@
+import { createFeatureSelector, createSelector } from '@ngrx/store';
+import { Trip } from '../../models/trips';
+import { TripState, tripsFeatureKey } from './reducer';
+
+export const selectFeature = createFeatureSelector<TripState>(tripsFeatureKey);
+
+const deriveDatesAndCost = (trip: Trip) => {
+  let startDate = new Date(8640000000000000);
+  let endDate = new Date(-8640000000000000);
```

Comment:
these numbers are a bit out there 😅 consider making the start date within the current month (or current day) since they can't plan for a trip that has already happened. And the end date maybe 10 years max from the current date, since it is rare for people to want to plan for a trip more than a decade out.

Reply:
So this is the smallest and largest possible date. Tim suggested doing it like this. There is always the chance that the app can be used years later so I wanted to have the dates as unreachable as possible in the past and future. Might be overkill but works perfectly 😅

Reply:
Definitely could be used decades in the future